zeroconf/__init__.py

```diff
@@ -2444,13 +2449,22 @@ def handle_response(self, msg: DNSIncoming) -> None:
             updated = True
 
             if record.unique:  # https://tools.ietf.org/html/rfc6762#section-10.2
-                for entry in self.cache.entries():
-
-                    if entry == record:
-                        updated = False
-
-                    if DNSEntry.__eq__(entry, record) and (record.created - entry.created > 1000):
-                        self.cache.remove(entry)
+                # Since the cache format is keyed on DNSEntry
+                # we can avoid iterating everything in the cache and
+                # only look though entries for the specific DNSEntry
+                # cache key.
+                entries = self.cache.entries_with_name(DNSEntry(record.name, record.type, record.class_).key)
+                if entries:
+                    # Make a copy since we are likely to delete
+                    # and we cannot iterate over something we
+                    # might remove
+                    for entry in entries.copy():
+
+                        if entry == record:
+                            updated = False
+
+                        if (record.created - entry.created > 1000) and DNSEntry.__eq__(entry, record):
```

Comment:
Is this optimization (reordering of the expressions) still necessary? If so please leave a comment mentioning this and include it in the commit message, otherwise it's not really visible that this changed.

Reply:
The > is still far cheaper than making the __eq__ call so will note that